flaml/automl.py

```diff
@@ -1108,10 +1108,22 @@ def _validate_data(
             from .data import DataTransformer
 
             self._transformer = DataTransformer()
+
             self._X_train_all, self._y_train_all = self._transformer.fit_transform(
                 X, y, self._state.task
             )
             self._label_transformer = self._transformer.label_transformer
+            if hasattr(self._label_transformer, "_label_list"):
+                for each_estimator in list(self._state.fit_kwargs_by_estimator.keys()):
+                    self._state.fit_kwargs_by_estimator[each_estimator].update(
+                        {"label_list": self._label_transformer._label_list}
+                    )
+            if self._state.task == TOKENCLASSIFICATION:
+                for each_estimator in list(self._state.fit_kwargs_by_estimator.keys()):
```

Comment:
What if self._state.fit_kwargs_by_estimator is empty and self._state.fit_kwargs is not? Remember that users can either set fit_kwargs_by_estimator or fit_kwargs

Reply:
addressed.